### DISCUSSION TITLE
Strange looking placement density heatmap

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3396] [General]
I would have expected the placement heatmap to extend to the top and to the right, but it seems like the heatmap stops at the grey area to the top and right...

Is OpenROAD doing the expected thing here?  If yes, what is the grey area?

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/5bcf1742-f124-41d4-a28e-05daf9fdfcf8)

The routing heatmap extends to the top and to the right, as I expected:

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/b99e0e03-55d6-4e1a-b5dc-a58708830736)

The DRC errors are routing congestion, not relevant to the question above, but it is what I was working on when I discovered the placement heatmap oddity. I think I need to run gui_cts and track down the buffer trees to see what is going on w.r.t routing concestion, though I think the pins could be spaced out and centered in the macro by the looks of it.

**Comment by QuantamHD** [comment]
The gray area is 0, and isn't strictly gray, it just looks that way since you have instance outlines on. Zero is a fully transparent box.

You get zero when an area is only filler/physical cells

**Reply by oharboe** [reply]
> The gray area is 0, and isn't strictly gray, it just looks that way since you have instance outlines on. Zero is a fully transparent box.
> 
> You get zero when an area is only filler/physical cells

I see.

How can I check that there are indeed only filler/physical cells in this area?

I am a little suspicious and would like to verify.

**Reply by QuantamHD** [reply]
Zoom in and click on the boxes and see what type they are. You can double click the placement density label on the left side, and make the grid finer as well
